tree_unique_args/src/loop_invariant.egg

```diff
@@ -0,0 +1,43 @@
+;; Loop Invariant
+(relation find-inv-Expr (Expr Expr))
+(relation find-inv-ListExpr (Expr ListExpr))
+
+;; bool: whether the term in the Expr is an invariant.
+(function is-inv-Expr (Expr Expr) bool :unextractable :merge (or old new))
```

Comment:
Why have an output bool here?
Why not have a relation instead, and match on the relation?

Reply:
The reason is for boundary analysis. In boundary analysis we also want to know if an Expr is not invariant (to find the boundary where an invariant's parent is not invariant), so in default I set all to be false and change when finding the invariant.